File: Data Structures & Algorithms/permutation-string/submission-0.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if (s1.size() > s2.size()) return false;
        vector<int> v1(126,0),v2(126,0);
        for(auto c: s1)v1[c]++;

        int l=0,m=s1.size(),n=s2.size();
        int i=0;
        while(i<(m-1)){
            v2[s2[i]]++;
            i++;
        }
        for(int r=m-1; r<n;r++){
            v2[s2[r]]++;
            if(v1==v2)return true;
            else{
                v2[s2[l]]--;
                l++;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if not root:
            return 0
        ls=[]
        def inorder(root):
            if not root:
                return
            inorder(root.left)
            ls.append(root.val)
            inorder(root.right)

        inorder(root)
        return ls[k-1]

